### ISSUE TITLE
Canonical Capsules: Unsupervised Capsules in Canonical Pose

### ISSUE THREAD

**Issue by AlexeyAB** [AlexeyAB/darknet#7105]
* https://canonical-capsules.github.io/
* https://arxiv.org/abs/2012.04718

> We propose an unsupervised capsule architecture for 3D point clouds. We compute capsule decompositions of objects through permutation-equivariant attention, and self-supervise the process by training with pairs of randomly rotated objects. Our key idea is to aggregate the attention masks into semantic keypoints, and use these to supervise a decomposition that satisfies the capsule invariance/equivariance properties. This not only enables the training of a semantically consistent decomposition, but also allows us to learn a canonicalization operation that enables object-centric reasoning. In doing so, we require neither classification labels nor manually-aligned training datasets to train. Yet, by learning an object-centric representation in an unsupervised manner, our method outperforms the state-of-the-art on 3D point cloud reconstruction, registration, and unsupervised classification. We will release the code and dataset to reproduce our results as soon as the paper is published.

![teaser](https://user-images.githubusercontent.com/4096485/101831974-c217f980-3b47-11eb-9d88-884297315e24.gif)
